sql3/test/defs/defs_date_functions.go

```diff
@@ -241,5 +241,49 @@ var datePartTests = TableTest{
 			),
 			Compare: CompareExactUnordered,
 		},
+		// dateTimeName tests
+		// make sure returning a string as the year still works
+		// if this works then the other parts converted to strings of digits should also work
+		{
+			SQLs: sqls(
+				"select _id, datetimename('yy', ts) from dateparttests",
+			),
+			ExpHdrs: hdrs(
+				hdr("_id", fldTypeID),
+				hdr("", fldTypeInt),
+			),
+			ExpRows: rows(
+				row(int64(1), "2012"),
+			),
+			Compare: CompareExactUnordered,
+		},
+		//check to make sure it gets a month name correctly
+		{
+			SQLs: sqls(
+				"select _id, datetimename('m', ts) from dateparttests",
+			),
+			ExpHdrs: hdrs(
+				hdr("_id", fldTypeID),
+				hdr("", fldTypeInt),
+			),
+			ExpRows: rows(
+				row(int64(1), "November"),
+			),
+			Compare: CompareExactUnordered,
+		},
+		// check to make sure it gets a day of the week correctly
+		{
+			SQLs: sqls(
+				"select _id, datetimename('w', ts) from dateparttests",
+			),
+			ExpHdrs: hdrs(
+				hdr("_id", fldTypeID),
+				hdr("", fldTypeInt),
+			),
+			ExpRows: rows(
+				row(int64(1), "Thursday"),
```

Comment:
I don't think this can possibly be correct, because you have a header saying fldTypeInt, and a string "Thursday".

I'm inferring that we maybe don't enforce the header column types, and maybe we should, but we should definitely not be claming that "Thursday" is an integer. Thursday is a man [Chesterton, 1908], and a man is not a number [McGoohan, 1967].